src/clib/pioc_support.c

```diff
@@ -148,35 +188,43 @@ void pio_log(int severity, const char *fmt, ...)
        many tabs before the message. */
     if (!severity)
     {
-        strcpy(ptr, ERROR_PREFIX);
+        strncpy(ptr, ERROR_PREFIX, (rem_len > 0) ? rem_len : 0);
```

Comment:
The coding style is that we don't add unnecessary code.

You may find your own unnecessary code to be easy to understand and modify, but I do not. So let's stick to only necessary code, then we will both (and all other C programmers) have the minimum of effort to understand the code.